src/Reference/HasOneSql.php

```diff
@@ -20,9 +20,10 @@ class HasOneSql extends HasOne
      *
      * Returns Expression in case you want to do something else with it.
      *
-     * @param string|Field|array $ourFieldName or [$field, ..defaults]
+     * @param string|Field|array $ourFieldName   or [$field, ..defaults]
+     * @param string|\Closure    $theirFieldName
```

Comment:
when we need Closure? It should be probably resolved before passing to this method

Reply:
We need the closure in this method in order to resolve `$theirModel` later.
The issue comes when a model is referencing its own class and there is usage of `addTitle` or `addField` in the init method which results in endless loop

Reply:
this needs to be solved definitelly different

Reply:
Ideas and code are welcome 😃

Reply:
the same issue is very common in Model init with circulars refs

this can be addressed in many ways (traverse owners and get what you want if already declared, stack trace, model registry, ...)

but definitelly not criple model design :)

Reply:
(In tests) it is solely related with addTitle/withTitle which is called from - from Model init :)

this is common issue not only related with title...